### DISCUSSION TITLE
[Dashboard] Support Filterbox validators

### DISCUSSION THREAD

**Discussion by cccs-jc** [apache/superset#14282] [Ideas]
**Describe the solution you'd like**

Ability to specify a validators inside a FilterBox.

We use Superset in the context of Security information and event management (SIEM). Our data often include IP values which users want to filter by adhoc IP ranges (using CIDR notation).

For example when the user inputs a filter value of 192.168.0.0/24 this filter is converted into a range query on the server side. We leverage jinja templating to convert the IP range into an SQL statement like 

`((IP_COLUMN >= lowIP) AND (IP_COLUMN <= highIP))`

This works quite well however there is no validation of the text provided by the user. Ideally we would like to validate the input in the client a bit like is done in the explorer UI.

It's possible to validate inputs in the explore UI since the viz populates the control panel and is free to customize the input control. However there is no way to specify a validator in a FilterBox.

Is this a feature being developed? Are there alternatives?

**Comment by junlincc** [comment]
@cccs-jc thanks for suggesting. we got this request quite often. the reason we are holding off on this is that FilterBox will be replaced by dashboard native filter https://github.com/apache/superset/issues/12148 as it rolls out. please follow step to enable the new feature and let us know if it doesn't work for you case. if not(likely), we are committed to make enhancement on the new solution.

**Comment by cccs-jc** [comment]
@junlincc it does not work in the native filters either. I'll will write a comment in the issue you mentioned. Thanks.
